apps/fhir/bluebutton/views/generic.py

```diff
@@ -151,6 +153,13 @@ def fetch_data(self, request, resource_type, *args, **kwargs):
 
         prepped = s.prepare_request(req)
 
+        if resource_type == 'Patient' and kwargs.get('resource_id') and prepped.headers.get('BlueButton-BeneficiaryId'):
+            # If it is a patient read request, confirm it is valid for the current user
+            # If not, throw a 404 before pinging BFD
+            if not valid_caller_for_patient_read(prepped.headers.get('BlueButton-BeneficiaryId'), kwargs.get('resource_id')):
```

Comment:
Lets pull prepped.headers.get('BlueButton-BeneId') out so it isn't duplicated in this block?